docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,192 @@
+
+`NEAR CLI` is built for:
+- NEAR `dApp developers`, who build smart-contracts, UIs, and tooling on NEAR.
+- `Tech-savvy` people automating their routines.
+- `Validators`
+
+`NEAR CLI UX principles`
+- All altering actions should have a confirmation step with an option to skip confirmation with an explicit command line parameter (e.g. `send` at the end of the command)
+- All direct children commands of a single parent command should be aligned (either represent an action or a resource, but never a mix of those on the same hierarchy level): `contract` -> `state` (resource) and `contract` -> `deploy` (action) are not aligned, so it should be either `contract` -> `get-state` + `contract` -> `deploy` or `contract` -> `state` -> `view` + `contract` -> `code` -> `deploy`
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+## `Core NEAR CLI` commands
+
+```
+account
+  - view-account-summary <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+ 
+  - create-subaccount <new-account-id> <initial-balance> 
+    (we will treat everything after the first dot in account id as the parent account (transaction signer))
+    - autogenerate-keypair network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - use-manually-provided-seed-prase "twelve words goes here" network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - use-manually-provided-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-account <account-id> beneficiary <beneficiary-account-id> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - list-keys <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+
+  - add-key <account-id>
+    - autogenerate-keypair network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - use-manually-provided-seed-prase "twelve words goes here" network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - use-manually-provided-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-key <account-id> <public-key> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - TODO: "import account" (aka "login") commands
+```
+
+```
+contract
+  - call-function
+    - as-read-only <account-id> <function-name> <function-args> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+    - as-transaction <account-id> <function-name> <function-args> --prepaid-gas <prepaid-gas> --attached-deposit <deposit-amount> network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
```

Comment:
@volovyk-s @BenKurrek @miraclx @ChaoticTempest How do you like this layout? I am quite happy about having a single "call-function" which has two clear options, where we will be able to teach the difference in great detail

Reply:
I like this a lot - great improvement.

Reply:
I like it!